### DISCUSSION TITLE
Issue using the MonoSyn tool

### DISCUSSION THREAD

**Discussion by GuilleSGCUPO** [petersenpeter/CellExplorer#96] [Q&A] [answered]
Hi guys, I'm trying to use Monosyn tool to check the monosynaptic connections. When I opened the CellMetrics in CellExplorer there were no connections between neurons displayed. However, if I run the script: 
`mono_res = ce_MonoSynConvClick(spikes,'includeInhibitoryConnections',true/false); % detects the monosynaptic connections
gui_MonoSyn(mono_res) % Shows the GUI for manual curation`
it appears the Monosyn window with every conection but it does not allow me to save the monosyn file.
If I open the CellExplorer tool and press "Adjust monosynaptic conections", it now allows me to save but Matlab drops this error, because there is no "putativeconections" variable inside the cellmetric struct:
>> cell_metrics = CellExplorer('metrics',cell_metrics);
Reference to non-existent field 'putativeConnections'.
Error in CellExplorer/adjustMonoSyn_UpdateMetrics (line 12445)
                        ia = ismember(cell_metrics.putativeConnections.excitatory(:,1), idx);
 Error using uiwait (line 81)
Error while evaluating Menu Callback.

Any idea on how can I fix it?
Thanks :)

**Comment by petersenpeter** [comment]
Hi

When you run the gui_MonoSyn gui, after making any changes and close the GUI, it should prompt you to save your changes to the mono_res structure. Is that not happening?

If no connections are detected, it will not show you the curation or save connection information in the metrics.

If you share your data, I can try to troubleshoot it.

**Reply by GuilleSGCUPO** [reply]
Hi,

Exactly, when I run the MonoSyn gui, either I make changes or not, when I close it, there is no message for saving the file. However, it generates an "ans" struct in the workspace with the corresponding info, so I can save it manually. 

Furthermore, about the "putativeConnections" issue I got, I came to the solution that if I run the loadCellMetrics.m function, it appears an "ans" struct with the variables about synaptic connections, so I can manually save it again in the cell_metrics struct. Although I guess it is not usual, right? However, it works :).

Guille

**Comment by petersenpeter** [comment, marked as answer]
You want to save the output in the correct parameter names:
`mono_res = gui_MonoSyn(mono_res)`
and
```
cell_metrics = ProcessCellMetrics;
cell_metrics = CellExplorer('metrics',cell_metrics); 
```

I think this was not clear from the documentation. In CellExplorer you can save your changes through the file menu. In gui_MonoSyn you want to do this manually if you first load mono_res into memory.

**Reply by GuilleSGCUPO** [reply]
Thank you so much Peter, that worked perfectly :)
